parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -66,6 +68,9 @@
   private Parquet() {
   }
 
+  private static Collection<String> READ_PROPERTIES_TO_REMOVE = Sets.newHashSet(
```

Comment:
Is this required because of Pig?  If yes, then should this have been fixed in Pig module rather than here?

Reply:
I wonder if this was because of Pig using the same job configuration across multiple instances of readers/writers.

Reply:
Yes, depending on the reader implementation, we found in some cases two different loaders in the same MR job would have properties that collide.  This is just to ensure the iceberg side doesn't get impacted.

Reply:
To be clear, this is when properties are copied out of the UDFContext and into the job configuration.  I don't believe we found any correctness issues on either the parquet pig or iceberg loader in terms of behavior, but the collision causes problems due to iceberg shading.

Reply:
We found this in Pig, but Iceberg should never try allow these to be set while reading because they are not used for Iceberg.

Reply:
Thanks for the explanation @danielcweeks and @rdblue!